crates/ruff_python_formatter/src/expression/expr_call.rs

```diff
@@ -88,12 +108,34 @@ impl FormatNodeRule<ExprCall> for FormatExprCall {
 impl NeedsParentheses for ExprCall {
     fn needs_parentheses(
         &self,
-        parenthesize: Parenthesize,
+        parent: AnyNodeRef,
         context: &PyFormatContext,
-    ) -> Parentheses {
-        match default_expression_needs_parentheses(self.into(), parenthesize, context) {
-            Parentheses::Optional => Parentheses::Never,
-            parentheses => parentheses,
+    ) -> OptionalParentheses {
+        self.func.needs_parentheses(parent, context)
```

Comment:
Never is incorrect. It requires parentheses if the func node requires parentheses (see attribute chain)

Reply:
yep i should have placed a todo comment about missing fluent style support there